rrdd/rrdd_server.ml

```diff
@@ -416,29 +417,51 @@ module Plugin = struct
 
 		(* Helper function to find plugin info from a uid. *)
 		let find ~(uid: P.uid) : plugin =
-			Mutex.execute registered_m
-				(fun () -> Hashtbl.find registered uid)
+			try
+				Mutex.execute registered_m (fun () -> Hashtbl.find registered uid)
+			with
+				Not_found ->
+					error "failed to find plugin %s" (P.string_of_uid uid);
+					raise Rrd_protocol.Read_error
+
+		let mute uid plugin =
+			Mutex.execute registered_m (fun () -> plugin.muted <- true);
+			warn "muting plugin %s" (P.string_of_uid uid)
+
+		let unmute uid plugin =
+			if plugin.muted then begin
+				Mutex.execute registered_m (fun () -> plugin.muted <- false);
+				warn "unmuting plugin %s" (P.string_of_uid uid)
+			end else
+				()
 
 		let get_payload ~(uid: P.uid) : Rrd_protocol.payload =
+			let plugin  = find uid in
 			try
-				let {reader} = find uid in
-				reader.Rrd_reader.read_payload ()
-			with e ->
-				let log e =
-					error "Failed to process plugin: %s (%s)"
-						(P.string_of_uid uid)
-						(Printexc.to_string e);
-					log_backtrace ()
-				in
-				let open Rrd_protocol in
-				match e with
-				| No_update -> raise No_update
-				| Invalid_header_string | Invalid_length | Invalid_checksum as e ->
-					log e;
-					raise e
-				| e ->
-					log e;
-					raise Read_error
+				let payload = plugin.reader.Rrd_reader.read_payload () in
+					unmute uid plugin;
+					payload
+			with
+				| e when not plugin.muted ->
+					mute uid plugin;
+					let log e =
+						error "Failed to process plugin: %s (%s)"
```

Comment:
As far as I understood, most of these errors are actually not errors. Why don't we use `warn` instead? It may hide some errors, but I think that whomever looks for rrdd plugins issues would still be able to easily get the informations.